lib/cli-engine.js

```diff
@@ -694,6 +690,8 @@ CLIEngine.prototype = {
                     descriptor.meta.hashOfConfig = hashOfConfig;
                     descriptor.meta.results = res;
                 }
+            } else {
```

Comment:
Sorry, was rushing to get to a meeting.  The line I linked to is also an `if (options.cache) {`, which happens before the file is processed.  I think the cache should probably be destroyed up there, rather than here after the call to `processFile`, so I was asking if there was a reason you put it down here instead.